packages/DataHeader/src/DataHeader.js

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Character from "@paprika/icon/lib/Character";
+import Numeric from "@paprika/icon/lib/Number";
+import DateTime from "@paprika/icon/lib/TimeAndDate";
+import ClockTime from "@paprika/icon/lib/ClockTime";
+import Date from "@paprika/icon/lib/Calendar";
+
+import * as sc from "./DataHeader.styles";
+import types from "./types";
+
+export default function DataHeader(props) {
+  const { renderActions, label, backgroundColor, color, type, icons, ...moreProps } = props;
+  const refHeader = React.useRef(null);
+
+  return (
+    <sc.Header
+      data-pka-anchor="DataHeader.Header"
+      ref={refHeader}
+      refHeader={refHeader}
+      $color={color}
+      $backgroundColor={backgroundColor}
+      {...moreProps}
+    >
+      <sc.Info data-pka-anchor="DataHeader.Info">
+        {icons && type && type in icons ? <sc.Icon data-pka-anchor="DataHeader.Icon">{icons[type]}</sc.Icon> : null}
+        <sc.Label data-pka-anchor="DataHeader.Label">{typeof label === "function" ? label() : label}</sc.Label>
+      </sc.Info>
+      {renderActions && typeof renderActions === "function" ? renderActions() : null}
+    </sc.Header>
+  );
+}
+
+DataHeader.types = types;
+DataHeader.propTypes = {
+  backgroundColor: PropTypes.string,
+  color: PropTypes.string,
+  label: PropTypes.oneOfType([PropTypes.func, PropTypes.string]).isRequired,
+  renderActions: PropTypes.func,
+  type: PropTypes.oneOf([types.type.TEXT, types.type.NUMERIC, types.type.DATE, types.type.DATE_TIME, types.type.TIME]),
+  icons: PropTypes.object, // eslint-disable-line
+};
+
+DataHeader.defaultProps = {
+  backgroundColor: null,
+  renderActions: null,
+  type: null,
+  icons: null,
+  color: null,
+};
+
+DataHeader.icons = {
+  [types.type.TEXT]: <Character />,
+  [types.type.NUMERIC]: <Numeric />,
+  [types.type.DATE]: <Date />,
+  [types.type.DATE_TIME]: <DateTime />,
+  [types.type.TIME]: <ClockTime />,
+};
```

Comment:
Do you still plan to extend the API for this component so that these icons can be customized?

Reply:
Also, what about Booleans? I know we made the Boolean icon ourselves, it should be in Wasabicons.